nixos/modules/programs/gamescope.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.gamescope;
+
+  gamescope = pkgs.gamescope.override {
+    extraWrapperArgs = optional (cfg.args != [ ])
+      ''--add-flags "${toString cfg.args}"''
+    ++ builtins.attrValues (mapAttrs (var: val: "--set-default ${var} ${val}") cfg.env);
+  };
+in
+{
+  options.programs.gamescope = {
+    enable = mkEnableOption "gamescope";
+
+    capSysNice = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Add cap_sys_nice capability to the GameScope binary.
+      '';
+    };
+
+    args = mkOption {
+      type = types.listOf types.string;
+      default = [ ];
+      example = [ "--rt" "--prefer-vk-device 8086:9bc4" ];
+      description = lib.mdDoc ''
+        Arguments passed to GameScope on startup.
+      '';
+    };
+
+    env = mkOption {
+      type = types.attrsOf types.string;
+      default = { };
+      example = {
```

Comment:
This example does seem pretty useful for people with nvidia laptops, maybe we could have an `exampleText` with a comment explaining what it does?